### ISSUE TITLE
Swipe-to-dismiss support improvement

### ISSUE THREAD

**Issue by alexbchr** [airbnb/epoxy#465]
Hi,

I recently wanted to be able to Swipe an item in the recyclerview to dismiss it. I also wanted the view to fade out as you swipe it away from the screen. I easily did so using the EpoxyTouchHelper class with the following code:
```
fun <T : EpoxyModel<*>> RecyclerView.initSwipe(clazz: Class<T>, onSwipeCompleted: () -> Unit) {
    EpoxyTouchHelper.initSwiping(this)
            .leftAndRight()
            .withTarget(clazz)
            .andCallbacks(object : EpoxyTouchHelper.SwipeCallbacks<T>() {
                override fun onSwipeCompleted(model: T, itemView: View?, position: Int, direction: Int) {
                    onSwipeCompleted()
                }

                override fun onSwipeProgressChanged(model: T, itemView: View, swipeProgress: Float, canvas: Canvas?) {
                    itemView.alpha = (1f - swipeProgress.absoluteValue)
                }
            })
}
```

I thought everything was working fine at first, but I then realized that if I cancel my swipe-to-dismiss so the view is not dismissed, the view's alpha value did not reset to 1. I therefore had the idea to implement `onSwipeReleased` like so:
```
override fun onSwipeReleased(model: T, itemView: View) {
    itemView.alpha = 1f
}
```
However, this also reset the alpha value when the view exceeded the swipe threshold to be dismissed, therefore causing it to suddenly turn completely opaque before leaving the screen.

Would it be possible to have an additional parameter passed to the onSwipeReleased callback function, like `boolean exceededDismissThreshold` or `boolean willBeDismissed`. Or maybe add another function like `onSwipeCancelled(model: T, itemView: View)`, which would be called after `onSwipeReleased`. This would allow people to animate the completion and cancellation of the swipe separately.

Thank you!

**Comment by elihart** [comment]
@alexbchr thanks for the well thought out post, this is definitely something I'd like to help better support.

I don't honestly use the swipe functionality myself (I built it for fun, hoping it would be useful to others), so I would heavily weigh your opinion on the best recommended approach here to support all use cases.

If we decide to pass an additional param we need to be ok with that being a breaking change to people that update.

Animating cancelation and release separately makes sense to me. Would you be willing to propose an API change via a PR?

**Comment by alexbchr** [comment]
I agree with you that adding a parameter would be a breaking change, so adding a `onSwipeCancelled` function may be the way to go. Especially since `SwipeCallbacks` is a class and not an interface, the implementation of `onSwipeCancelled` would be optional to the user.

@elihart I'll try to do a PR with this change in the next week or so.

I'll also try to leverage the functionality if I think of any use case that would need a breaking change.

**Comment by elihart** [comment]
Sounds great, thanks!
